### PR TITLE
chore: format codebase with Prettier

### LINKED ISSUE #226: Format codebase with Prettier and check in CI everything is formatted

### PR DESCRIPTION
**Pull request by anton-trunov**
and introduce a CI check that the codebase is properly formatted

Closes #226

* [x] I have updated CHANGELOG.md
* [x] I have added unit tests to demonstrate the contribution is correctly implemented
* [x] I have run all the tests locally and no test failure was reported
* [x] I did not introduce unrelated formatting/whitespace changes in the code base
* [x] I did not do unrelated and/or undiscussed refactorings

Everything is the default, except I kept the 4 spaces indentation level for TS and JS files (see `.editorconfig`) and excluded the auto-generated and vendored files that are not in `.gitignore`, which Prettier ignores as well (see `.prettierignore`).

### CODE REVIEW

File: .prettierignore
Comment (novusnota) [inline .prettierignore:1]: Here it's better to prepend all those paths with `/` to ensure that Prettier ignores only those and not any relative `grammar/` folders, for example.

**Review by novusnota — CHANGES_REQUESTED** [review]

File: .prettierignore
Comment (anton-trunov) [reply .prettierignore:1]: should be fixed now

Comment (novusnota) [reply .prettierignore:1]: I meant the following:

```gitignore
/dist

/src/func/funcfiftlib.js
/src/func/funcfiftlib.wasm.js
/src/grammar/grammar.ohm-bundle.d.ts
/src/grammar/grammar.ohm-bundle.js
/src/imports/stdlib.ts
/grammar
````

See: https://git-scm.com/docs/gitignore#_examples

**Review by byakuren-hijiri — DISMISSED** [review]
I like the idea of unified code formatting, and the default configuration suggested by Prettier looks good to me.

I would suggest only removing the following line from the`pull_request_template.md`, as it is checked by CI for now:

>  I did not introduce unrelated formatting/whitespace changes in the code base

File: .prettierignore
Comment (anton-trunov) [reply .prettierignore:1]: thanks, never learnt that actually before now :)

**Comment by anton-trunov** [comment]
@byakuren-hijiri thanks! [if only there was a way to automatically check all the other bullet points, lol]

**Review by novusnota — APPROVED** [review]
Neat!
"Live and learn ~~and break CI~~" :)

**Pull request merged**